rust-bins/wallet-notes/README.md

```diff
@@ -152,6 +152,119 @@ The returned value is a JSON object with the following fields:
 An example input to this request is in the file [create_transfer-input.json](files/create_transfer-input.json).
 An example output to this request is in the file [create_transfer-output.json](files/create_transfer-output.json).
 
+## create_configure_delegation_transaction
+
+Semantics: Creates a `configure delegation`-transaction with the provided values.
+
+This function takes as input a NUL-terminated UTF8-encoded string. The string
+must be a valid JSON object with fields
+
+- `"from"` ... address of the sender account.
+
+- `"expiry"` ... unix timestamp of the expiry date of the transaction.
+
+- `"nonce"` ... nonce of the sender account.
+
+- `"keys"` ... mapping with the keys of the sender account.
+
+- `"energy"` ... max energy wanted for the transfer.
+
+The following fields are optional:
+
+- `"capital"` ... string containing the amount to be staked.
+
+- `"restakeEarnings"` ... bool indicating whether earnings should be restaked.
+
+- `"delegationTarget"` ... JSON indicating either delegation to the L-pool or to a baker pool.
+
+The delegation target should either be of the form
+```json
+{
+    "type": "delegateToLPool"
+}
+```
+or
+
+```json
+{
+    "type": "delegateToBaker",
+    "targetBaker": 100
+}
+```
+where `100` should be replaced with relevant baker id.
+
+To add a delegator, all of the optional fields must be present. For an existing delegator the fields that are present will be updated on chain. A delegator is removed if the `capital` is set to `"0"`.
+
+The returned value is a JSON object with the following fields:
+
+- `"signatures"` ... list with signatures of the transaction with the provided keys.
+
+- `"transaction"` ... the serialized transaction that can be sent to the chain.
+
+Example input to this request are in the files [create_configure_delegation_transacion-input.json](files/create_configure_delegation_transacion-input.json) and [2-create_configure_delegation_transacion-input.json](2-files/create_configure_delegation_transacion-input.json).
+Example output to this request are in the files [2-files/create_configure_delegation_transacion-output.json](files/create_configure_delegation_transacion-output.json).
+
+## create_configure_baker_transaction
+
+Semantics: Creates a transfer transaction with the provided values.
+
+This function takes as input a NUL-terminated UTF8-encoded string. The string
+must be a valid JSON object with fields
+
+- `"from"` ... address of the sender account.
+
+- `"expiry"` ... unix timestamp of the expiry date of the transaction.
+
+- `"nonce"` ... nonce of the sender account.
+
+- `"keys"` ... mapping with the keys of the sender account.
+
+- `"energy"` ... max energy wanted for the transfer.
+
+The following fields are optional:
+
+- `"capital"` ... string containing the amount to be staked.
+
```

Comment:
Missing: `"restakeEarnings"` and `"url"`. The length limit on the url should be stated.

Reply:
Done.